query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/relation/CanAddRelation.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.querybuilder.relation;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.querybuilder.QueryBuilderDsl;
+import com.datastax.oss.driver.api.querybuilder.term.Term;
+import com.datastax.oss.driver.internal.querybuilder.DefaultRaw;
+import com.datastax.oss.driver.internal.querybuilder.relation.CustomIndexRelation;
+import com.datastax.oss.driver.internal.querybuilder.relation.DefaultColumnComponentRelationBuilder;
+import com.datastax.oss.driver.internal.querybuilder.relation.DefaultColumnRelationBuilder;
+import com.datastax.oss.driver.internal.querybuilder.relation.DefaultTokenRelationBuilder;
+import com.datastax.oss.driver.internal.querybuilder.relation.DefaultTupleRelationBuilder;
+import com.google.common.collect.Iterables;
+import java.util.Arrays;
+
+/** A statement that is ready to accept relations in its WHERE clause. */
+public interface CanAddRelation<SelfT extends CanAddRelation<SelfT>> {
```

Comment:
When we add UPDATE and DELETE statements, they only need to implement this interface and the two `where()` methods, and they will get all the fluent shortcuts for free.